presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ClusterConfigAPI.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller.api;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.teradata.prestomanager.common.ExtendedStatus;
+import com.teradata.prestomanager.common.NodeSet;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+import javax.inject.Singleton;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import java.net.URI;
+
+import static com.teradata.prestomanager.common.RequestBuilder.newRequestBuilder;
+
+@Path("/cluster/config")
+@Api(description = "the config API for all nodes in cluster")
+@Singleton
+public final class ClusterConfigAPI
+{
+    private NodeSet nodeSet = NodeSet.getInstance();
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @ApiOperation(value = "Get available configuration files")
+    @ApiResponses(value = {
+            @ApiResponse(code = 207, message = "Retrieved configuration", response = String.class)})
+    public synchronized Response getConfig()
+    {
+        ImmutableList.Builder<Response> responseList = ImmutableList.builder();
+        UriBuilder uriBuilder = UriBuilder.fromUri("/config");
```

Comment:
We will have that class for string constants. It would involve changing the Agent API as well, so probably would be better to leave that for next patch.